sv_pipeline/load_data.py

```diff
@@ -0,0 +1,392 @@
+            if parsed and out_file:
+                out_file.write(line)
+
+    if out_file:
+        out_file.close()
+
+
+def subset_and_group_svs(input_dataset, sample_subset, sample_remap, ignore_missing_samples, write_subsetted_bed=False):
+    parsed_svs_by_name = {}
+    found_samples = set()
+    skipped_samples = set()
+    out_file_path = None
+    if write_subsetted_bed:
+        file_path = input_dataset.split('/')
+        file_path[-1] = 'subset_{}'.format(file_path[-1])
+        out_file_path ='/'.join(file_path)
+
+    def _parse_row(row, header_indices):
+        sample_id = get_field_val(row, SAMPLE_COL, header_indices)
+        if sample_remap and sample_id in sample_remap:
+            sample_id = sample_remap[sample_id]
+        if sample_subset is None or sample_id in sample_subset:
+            parse_sv_row(row, parsed_svs_by_name, header_indices, sample_id)
+            found_samples.add(sample_id)
+            return True
+        else:
+            skipped_samples.add(sample_id)
+            return False
+
+    load_file(input_dataset, _parse_row, out_file_path=out_file_path)
+
+    print('Found {} sample ids'.format(len(found_samples)))
+    if sample_subset:
+        if len(found_samples) != len(sample_subset):
+            missed_samples = sample_subset - found_samples
+            missing_sample_error = 'Missing the following {} samples:\n{}'.format(
+                len(missed_samples), ', '.join(sorted(missed_samples))
+            )
+            if ignore_missing_samples:
+                print(missing_sample_error)
+            else:
+                print('Samples in callset:\n{}'.format(', '.join(sorted(skipped_samples))))
+                raise Exception(missing_sample_error)
+
+    return parsed_svs_by_name
+
+
+def add_in_silico(svs_by_variant_id, file_path):
+    def _parse_row(row, header_indices):
+        variant_id = get_variant_id(row, header_indices)
+        if variant_id in svs_by_variant_id:
+            svs_by_variant_id[variant_id].update(get_parsed_column_values(row, header_indices, [IN_SILICO_COL]))
+
+    load_file(file_path, _parse_row, columns=IN_SILICO_COLS)
+
+
+def format_sv(sv):
+    sv[TRANSCRIPTS_FIELD] = [{'gene_id': gene} for gene in sv[GENES_FIELD]]
+    sv['transcriptConsequenceTerms'] = [sv[CALL_FIELD]]
+    if sv[SF_FIELD]:
+        sv['sn'] = int(sv[SC_FIELD] / sv[SF_FIELD])
+    sv['pos'] = sv[START_COL]
+    sv['xpos'] = CHROM_TO_XPOS_OFFSET[sv[CHROM_FIELD]] + sv[START_COL]
+    sv['xstart'] = sv['xpos']
+    sv['xstop'] = CHROM_TO_XPOS_OFFSET[sv[CHROM_FIELD]] + sv[END_COL]
+    sv['samples'] = []
+    for genotype in sv[GENOTYPES_FIELD]:
+        sample_id = genotype['sample_id']
+        sv['samples'].append(sample_id)
+
+        cn_key = 'samples_cn_{}'.format(genotype['cn']) if genotype['cn'] < 4 else 'samples_cn_gte_4'
+        if cn_key not in sv:
+            sv[cn_key] = []
+        sv[cn_key].append(sample_id)
+
+        if genotype[QS_FIELD] > 1000:
```

Comment:
I can't remember, is 1000 the cutoff for QS from the Talkowski group where the score is no longer informative?

Reply:
This is actually based on a slack conversation I had with you while I was writing the first pass at this pipeline :) : https://the-tgg.slack.com/archives/GUVVDL6MQ/p1583964252004600

Reply:
LOL sorry memory of a goldfish lately...